src/Og.php

```diff
@@ -446,4 +449,36 @@ public static function getSelectionHandler($entity, $bundle, $field_name, array
     return \Drupal::service('plugin.manager.entity_reference_selection')->createInstance('og:default', $options);
   }
 
+  /**
+   * Creates and saves an OG membership.
+   *
+   * @param \Drupal\Core\Entity\FieldableEntityInterface $entity
+   * @param \Drupal\Core\Entity\EntityInterface $group
+   * @param string $membership_type
+   * @param string $field_name
+   *
+   * @return \Drupal\og\Entity\OgMembership
+   */
+  public static function createMembership(FieldableEntityInterface $entity, EntityInterface $group, $membership_type = OG_MEMBERSHIP_TYPE_DEFAULT, $field_name = OG_AUDIENCE_FIELD) {
+    $field_definition = $entity->getFieldDefinition($field_name);
+
+    if (!$field_definition) {
```

Comment:
I think that maybe we should also have these two checks in `Drupal\og\Entity\OgMembership::preSave` as well